File: app/src/main/java/org/deiverbum/app/ui/fragments/OracionesFragment.java
package org.deiverbum.app.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import org.deiverbum.app.R;
import org.deiverbum.app.data.adapters.OracionItem;
import org.deiverbum.app.data.adapters.OracionesAdapter;
import org.deiverbum.app.databinding.FragmentOracionesBinding;
import org.deiverbum.app.utils.Utils;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/**
 * <p>
 * Este Fragmento maneja lo relativo al módulo <b>Oraciones</b>.
 * Muestra un <code>RecyclerView</code> con las diferentes oraciones
 * de la Liturgy de las Horas.
 * </p>
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2022.1
 */
public class OracionesFragment extends Fragment {
    RecyclerView recyclerView;
    private FragmentOracionesBinding binding;
    private List<OracionItem> mList;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        binding = FragmentOracionesBinding.inflate(inflater, container, false);
        View root = binding.getRoot();
        recyclerView = binding.recyclerView;
        mList = new ArrayList<>();
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        recyclerView.setAdapter(new OracionesAdapter(mList));
        prepareItems();
        pickOutDate();

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    private void prepareItems() {
        mList.add(new OracionItem(1, "Misterios Gloriosos", "Domingos y Miércoles", R.id.nav_rosario));
        mList.add(new OracionItem(2, "Misterios Gozosos", "Lunes y Sábados", R.id.nav_rosario));
        mList.add(new OracionItem(3, "Misterios Dolorosos", "Martes y Viernes", R.id.nav_rosario));
        mList.add(new OracionItem(4, "Misterios Luminosos", "Jueves", R.id.nav_rosario));
        mList.add(new OracionItem(5, "Letanías Lauretanas", "Solamente las Letanías", R.id.nav_rosario));
        mList.add(new OracionItem(6, "Ángelus", "Recuerda la Encarnación de Cristo", R.id.nav_rosario));
        mList.add(new OracionItem(7, "Regina Coeli", "En lugar del Àngelus, en el tiempo de Pascua", R.id.nav_rosario));
        mList.add(new OracionItem(8, "Via Crucis 2003", "Con meditaciones de Juan Pablo II", R.id.nav_rosario));
        mList.add(new OracionItem(9, "Via Crucis 2005", "Con meditaciones de Joseph Ratzinger", R.id.nav_rosario));
        //mAdapter.notifyDataSetChanged();
    }

    private void pickOutDate() {
        Bundle bundle = getArguments();
        String dateString;
        if (bundle != null) {
            dateString = bundle.getString("FECHA") == null ? Utils.getHoy() : bundle.getString("FECHA");
        } else {
            dateString = Utils.getHoy();
        }
        ActionBar actionBar = ((AppCompatActivity) requireActivity()).getSupportActionBar();
        Objects.requireNonNull(actionBar).setSubtitle(Utils.getTitleDate(dateString));
    }

}

File: app/src/main/java/org/deiverbum/app/core/presentation/oraciones/OracionesFragment.kt
package org.deiverbum.app.core.presentation.oraciones

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.viewbinding.ViewBinding
import dagger.hilt.android.AndroidEntryPoint
import org.deiverbum.app.R
import org.deiverbum.app.core.presentation.base.BaseHomeFragment
import org.deiverbum.app.core.presentation.oraciones.adapter.OracionItem
import org.deiverbum.app.core.presentation.oraciones.adapter.OracionesAdapter
import org.deiverbum.app.databinding.FragmentOracionesBinding
import org.deiverbum.app.util.Constants.FILE_ANGELUS
import org.deiverbum.app.util.Constants.FILE_LITANIES
import org.deiverbum.app.util.Constants.FILE_REGINA
import org.deiverbum.app.util.Constants.FILE_ROSARY
import org.deiverbum.app.util.Constants.FILE_VIA_CRUCIS_2003
import org.deiverbum.app.util.Constants.FILE_VIA_CRUCIS_2005


@AndroidEntryPoint
class OracionesFragment : BaseHomeFragment<FragmentOracionesBinding>() {
    private lateinit var mAdapter: OracionesAdapter
    private lateinit var mList: List<OracionItem>

    override fun constructViewBinding(): ViewBinding =
        FragmentOracionesBinding.inflate(layoutInflater)

    override fun init(viewBinding: ViewBinding) {
        prepareItems()
        initUi()
    }

    private fun initUi() {
        getViewBinding().recyclerView.run {
            val actionBar = (requireActivity() as AppCompatActivity).supportActionBar
            actionBar?.subtitle = ""
            layoutManager = LinearLayoutManager(context)
            adapter = mAdapter
        }
    }

    private fun prepareItems() {
        mList = ArrayList()
        mList = listOf(
            OracionItem(
                1,
                "Misterios Gloriosos",
                "Domingos y Miércoles",
                R.id.nav_rosario,
                FILE_ROSARY
            ),
            OracionItem(
                2,
                "Misterios Gozosos",
                "Lunes y Sábados",
                R.id.nav_rosario,
                FILE_ROSARY
            ),
            OracionItem(
                3,
                "Misterios Dolorosos",
                "Martes y Viernes",
                R.id.nav_rosario,
                FILE_ROSARY
            ),
            OracionItem(4, "Misterios Luminosos", "Jueves", R.id.nav_rosario, "raw/rosario.json"),
            OracionItem(
                5,
                "Letanías Lauretanas",
                "Solamente las Letanías",
                R.id.nav_letanias,
                FILE_LITANIES
            ),
            OracionItem(6, "Ángelus", "Excepto Pascua", R.id.nav_angelus, FILE_ANGELUS),
            OracionItem(7, "Regina Coeli", "Tiempo de Pascua", R.id.nav_regina, FILE_REGINA),
            OracionItem(
                8,
                "Via Crucis 2003",
                "Juan Pablo II",
                R.id.nav_viacrucis2003,
                FILE_VIA_CRUCIS_2003
            ),
            OracionItem(
                9,
                "Via Crucis 2005",
                "Benedicto XVI",
                R.id.nav_viacrucis2005,
                FILE_VIA_CRUCIS_2005
            )
        )

        mAdapter = OracionesAdapter(mList)
    }
}